src/cortxfs/cortxfs_ops.c

```diff
@@ -614,19 +617,28 @@ int cfs_rename(struct cfs_fs *cfs_fs, cfs_cred_t *cred,
 	bool overwrite_dst = false;
 	bool rename_inplace = false;
 	bool is_dst_non_empty_dir = false;
-	struct stat *stat = NULL;
-	cfs_ino_t sino;
-	cfs_ino_t dino;
 	str256_t k_sname;
 	str256_t k_dname;
 	mode_t s_mode = 0;
 	mode_t d_mode = 0;
+	cfs_ino_t *sdir_ino = NULL;
```

Comment:
cfs_rename( ) should be refactored, it's almost 200 lines and becoming unreadable. You can either add a todo here, or do it as part of this patch.  Alternatively, this function would again be modified when the signature would be changed, i.e when the file handles would be passed instead of inodes. The refactoring could be done then.

Reply:
As per me, I have done my best to remove unnecessary part but as there are so many case hence so many variables and so it seems complex. So this can be broken down further in function but not sure about how better it can be optimized further. Will take up this later on as it needed further investigation and it can be taken up as separate task after all activities of FH are completed. As of now adding comment in code.